airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/documentation.py

```diff
@@ -101,31 +115,299 @@ def _run(self, connector: Connector) -> CheckResult:
+            # any check is True, indexes didn't move to the next step
+            errors.append(reason_titles_not_match(heading_names_value, template_headings_value, template_headings))
+            return errors
+        # indexes didn't move to the last required one, so some headers are missed
+        if template_headings_index != template_headings_len:
+            errors.append(reason_missing_titles(template_headings_index, template_headings, non_required_heading))
+            return errors
+
         return errors
 
-    def check_sections(self, doc_lines: List[str]) -> List[str]:
+    def _run(self, connector: Connector) -> CheckResult:
+        if not connector.documentation_file_path or not connector.documentation_file_path.exists():
+            return self.fail(
+                connector=connector,
+                message="Could not check documentation structure as the documentation file is missing.",
+            )
+
+        doc_lines = [line.lower() for line in connector.documentation_file_path.read_text().splitlines()]
+        if not doc_lines:
+            return self.fail(
+                connector=connector,
+                message="Documentation file is empty",
+            )
+
+        docs_content = connector.documentation_file_path.read_text().rstrip()
+        errors = self.check_docs_structure(docs_content, connector.name_from_metadata)
+
+        if errors:
+            return self.fail(
+                connector=connector,
+                message=f"Documentation headers ordering/naming doesn't follow guidelines:\n {'. '.join(errors)}",
+            )
+        return self.pass_(
+            connector=connector,
+            message="Documentation guidelines are followed",
+        )
+
+
+class CheckDocumentationDescriptions(CheckDocumentationContent):
+    name = "Connectors documentation descriptions follows our guidelines"
+    description = f"The user facing connector documentation should follow the guidelines defined in the [documentation standards]({consts.DOCUMENTATION_STANDARDS_URL})."
+
+    PREREQUISITES = "Prerequisites"
+    HEADING = "heading"
+    CREDENTIALS_KEYWORDS = ["account", "auth", "credentials", "access", "client"]
+
+    def _headings_description(self, connector_name: str) -> dict[str, Path]:
+        """
+        Headings with path to file with template description
+        """
+        descriptions_paths = {
+            connector_name: Path(__file__).parent / "doc_templates/source.txt",
+            "For Airbyte Cloud:": Path(__file__).parent / "doc_templates/for_airbyte_cloud.txt",
+            "For Airbyte Open Source:": Path(__file__).parent / "doc_templates/for_airbyte_open_source.txt",
+            "Supported sync modes": Path(__file__).parent / "doc_templates/supported_sync_modes.txt",
+            "Tutorials": Path(__file__).parent / "doc_templates/tutorials.txt",
+            "Changelog": Path(__file__).parent / "doc_templates/changelog.txt",
+        }
```

Comment:
I'd highly recommend using a single Jinja template which we could use to:
* Pre-generate documentation with the correct structure for new connectors 
* Help developers figure what's the expected structure

Reply:
jinja template was added. also I used it to generate descriptions for some checks, so it now get example from template.md file and puts it to qa-checks doc